pkg/engine2/operational_eval/vertex_property_test.go

```diff
@@ -113,3 +113,103 @@ func Test_propertyVertex_evaluateEdgeOperational(t *testing.T) {
 		})
 	}
 }
+
+func Test_propertyVertex_Dependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	dcap := NewMockdependencyCapturer(ctrl)
+
+	type fields struct {
+		Ref           construct.PropertyRef
+		Template      knowledgebase.Property
+		EdgeRules     map[construct.SimpleEdge][]knowledgebase.OperationalRule
+		ResourceRules map[string][]knowledgebase.OperationalRule
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		mocks   func()
+		wantErr bool
+	}{
+		{
+			name: "property vertex with template",
+			fields: fields{
+				Ref: construct.PropertyRef{
+					Property: "test",
+					Resource: construct.ResourceId{Name: "test"},
+				},
+				Template: &properties.StringProperty{
+					PropertyDetails: knowledgebase.PropertyDetails{
+						OperationalRule: &knowledgebase.PropertyRule{
+							If: "test",
+						},
+					},
+				},
+			},
+			mocks: func() {
+				dcap.EXPECT().ExecutePropertyRule(knowledgebase.DynamicValueData{
+					Resource: construct.ResourceId{Name: "test"},
+				}, knowledgebase.PropertyRule{
+					If: "test",
+				}).Return(nil)
+			},
+		},
+		{
+			name: "property vertex with edge rules and resource rules",
+			fields: fields{
+				Ref: construct.PropertyRef{
+					Property: "test",
+					Resource: construct.ResourceId{Name: "test"},
+				},
+				EdgeRules: map[construct.SimpleEdge][]knowledgebase.OperationalRule{
+					{
+						Source: construct.ResourceId{Name: "test"},
+						Target: construct.ResourceId{Name: "test2"},
+					}: {
+						{
+							If: "testE",
+						},
+					},
+				},
+				ResourceRules: map[string][]knowledgebase.OperationalRule{
+					"test": {
+						{
+							If: "testR",
+						},
+					},
+				},
+			},
+			mocks: func() {
+				dcap.EXPECT().ExecuteOpRule(knowledgebase.DynamicValueData{
+					Resource: construct.ResourceId{Name: "test"},
+				}, knowledgebase.OperationalRule{
+					If: "testR",
+				}).Return(nil)
+				dcap.EXPECT().ExecuteOpRule(knowledgebase.DynamicValueData{
+					Resource: construct.ResourceId{Name: "test"},
+					Edge:     &graph.Edge[construct.ResourceId]{Source: construct.ResourceId{Name: "test"}, Target: construct.ResourceId{Name: "test2"}},
+				}, knowledgebase.OperationalRule{
+					If: "testE",
+				}).Return(nil)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			prop := &propertyVertex{
+				Ref:           tt.fields.Ref,
+				Template:      tt.fields.Template,
+				EdgeRules:     tt.fields.EdgeRules,
+				ResourceRules: tt.fields.ResourceRules,
+			}
```

Comment:
This is a little awkward, why not just have `tt` have a `prop propertyVertex` field? Or some other nicer ways to slim down the test cases (like using strings instead of IDs / edges / etc).